File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TJSONPair.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TJSONPair.h"



@implementation TJSONPair
@synthesize name,value;

-(id) initWithName:(NSString* )aName andValue:(TJSONValue*)aValue{
	self = [self init];
	if (self) {
		self.name = aName;
		self.value = aValue;
	}
	return self;

}



@end


@implementation TJSONPair(JSONPairCreation)
+(id) JSONPairWithName:(NSString* )aName andValue:(TJSONValue*)aValue{
	return [[TJSONPair alloc]initWithName:aName andValue:aValue];

}
@end

File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TJSONPair.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * Represents json pair objects. A JSONPair is an object that has a name an
 * value that represents a {@link TJSONValue}.
 * 
 */
public class TJSONPair {

	public String name;
	public TJSONValue value;

	/**
	 * Class constructor, initializes the TJSONPair with a TJSONValue with the
	 * specified name.
	 * 
	 * @param name
	 * @param value
	 */
	public TJSONPair(String name, TJSONValue value) {
		this.name = name;
		this.value = value;
	}

	/**
	 * Initializes the TJSONPair with a String with the specified name.
	 * 
	 * @param name
	 * @param value
	 */
	public TJSONPair(String name, String value) {
		this.name = name;
		this.value = new TJSONString(value);
	}

}
